depends/patches/proton/minimal-build.patch

```diff
@@ -0,0 +1,288 @@
+From 03f5fc0826115edbfca468261b70c0daf627f488 Mon Sep 17 00:00:00 2001
+From: Simon <simon@bitcartel.com>
+Date: Thu, 27 Apr 2017 17:15:59 -0700
+Subject: [PATCH] Only build minimal dependencies needed for cpp bindings.
+
+---
+ CMakeLists.txt                            | 13 +++++++------
+ examples/cpp/CMakeLists.txt               |  1 +
+ proton-c/CMakeLists.txt                   | 32 +++++++++++++++----------------
+ proton-c/bindings/CMakeLists.txt          |  6 +++---
+ proton-c/bindings/cpp/CMakeLists.txt      | 24 +++++++++++------------
+ proton-c/bindings/cpp/docs/CMakeLists.txt |  2 +-
+ proton-c/docs/api/CMakeLists.txt          |  2 +-
+ 7 files changed, 41 insertions(+), 39 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index b538ffd..4a5e787 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -18,14 +18,15 @@
+ #
+ cmake_minimum_required (VERSION 2.8.7)
+ 
++set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+ project (Proton C)
+ 
+ # Enable C++ now for examples and bindings subdirectories, but make it optional.
+ enable_language(CXX OPTIONAL)
+ 
+ # Enable testing
+-enable_testing()
+-include (CTest)
++#enable_testing()
++#include (CTest)
+ 
+ # Pull in local cmake modules
+ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/tools/cmake/Modules/")
+@@ -141,7 +142,7 @@ set (BINDINGS_DIR ${LIB_INSTALL_DIR}/proton/bindings)
+ 
+ set (SYSINSTALL_BINDINGS OFF CACHE BOOL "If SYSINSTALL_BINDINGS is OFF then proton bindings will be installed underneath ${BINDINGS_DIR} and each user will need to modify their interpreter configuration to load the appropriate binding. If SYSINSTALL_BINDINGS is ON, then each language interpreter will be queried for the appropriate directory and proton bindings will be installed and available system wide with no additional per user configuration.")
+ 
+-set (BINDING_LANGS PERL PHP PYTHON RUBY)
++#set (BINDING_LANGS PERL PHP PYTHON RUBY)
+ 
+ foreach (LANG ${BINDING_LANGS})
+   set (SYSINSTALL_${LANG} OFF CACHE BOOL "Install ${LANG} bindings into interpreter specified location.")
+@@ -156,10 +157,10 @@ set (PROTON_SHARE ${SHARE_INSTALL_DIR}/proton-${PN_VERSION})
+ # End of variables used during install
+ 
+ # Check for valgrind here so tests under proton-c/ and examples/ can use it.
+-find_program(VALGRIND_EXE valgrind DOC "Location of the valgrind program")
++#find_program(VALGRIND_EXE valgrind DOC "Location of the valgrind program")
+ mark_as_advanced (VALGRIND_EXE)
```

Comment:
I think you mean that you want "#mark_as_advanced" ?  From cmake docs, 'mark_as_advanced' doesn't do anything with regards to the build, it will hide the variable from the CMake GUI.   I didn't comment it out as I wanted the patch to be as small as possible.